### ISSUE TITLE
mux: use meow_common::atomic::AtomicU — AtomicU64 breaks the MIPS32 build

### ISSUE THREAD

**Issue by madeye** [madeye/mihomo-rust#421]
The `mux` module uses `std::sync::atomic::AtomicU64` directly instead of the
workspace's platform-adaptive alias `meow_common::atomic::AtomicU`.

Sites:
- `crates/meow-proxy/src/mux/client.rs:22,300,544` — `MuxSession::last_used_ms`
- `crates/meow-proxy/src/mux/packet.rs` and `mux/muxcool.rs` (test constructors)

`mipsel-unknown-linux-musl` has no 64-bit atomics, so `AtomicU64` does not exist there:

```
$ rustc --print cfg --target mipsel-unknown-linux-musl | grep atomic
target_has_atomic="16"
target_has_atomic="32"
target_has_atomic="8"
target_has_atomic="ptr"
```

`mux` is in `meow-proxy`/`meow-config` `default` and therefore in `meow-app`'s
`default = ["full", "boring-tls"]`, so the whole default build fails to compile
for that target. Not caught by CI because the current release matrix (b4c7006)
contains no 64-bit-atomic-less target, but mipsel is still a named ADR-0007
size target and this is the exact failure fixed once before in
5ef7636 *"fix(anytls): don't assume 64-bit atomics (mips32 release build)"*.

**Fix:** switch to `meow_common::atomic::{AtomicU, Uint}`, and follow the alias
module's rule for the millisecond clock — compare in the truncated domain with
`wrapping_sub`, never widen to `u64` first (see `UdpSession::idle_for`).

**Related:** `MuxClient::now_ms()` reads `SystemTime`, so a backwards wall-clock
step makes `saturating_sub` return 0 and defers idle eviction. Moving to a
monotonic base would fix both issues at once.

Follow-up from the review of the #412 mux stack.

**Comment by madeye** [comment]
@aimkiray — owner, please, as follow-up to #414 (`mux/client.rs`). This is the highest-priority item of the seven: `mipsel-unknown-linux-musl` currently fails to compile with the `mux` feature, which is on by default. The fix is a type-alias swap to `meow_common::atomic::{AtomicU, Uint}`, with the `wrapping_sub` comparison rule for the ms clock. 5ef7636 is the precedent from the anytls case.
